cpp/src/io/json/json_column.cu

```diff
@@ -410,12 +410,38 @@ std::pair<std::unique_ptr<column>, std::vector<column_name_info>> device_json_co
       std::vector<std::unique_ptr<column>> child_columns;
       std::vector<column_name_info> column_names{};
       size_type num_rows{json_col.num_rows};
+      if (schema.has_value() and schema.value().column_order.has_value()) {
+        CUDF_EXPECTS(schema.value().child_types.size() == schema.value().column_order->size(),
+                     "Input schema column order size mismatch with input schema child types");
+      }
```

Comment:
Use `validate_column_order`?

Reply:
Or probably we can remove this completely, since the schema has already been validated when it was set.